src/main/java/org/opensearch/ad/transport/ForwardADTaskTransportAction.java

```diff
@@ -125,6 +151,12 @@ protected void doExecute(Task task, ForwardADTaskRequest request, ActionListener
                     adTaskCacheManager.removeHistoricalTaskCache(detectorId);
                 } else {
                     adTaskCacheManager.removeRealtimeTaskCache(detectorId);
+                    // If hash ring changed like new node added when scale out, the realtime job coordinating node may
+                    // change, then we should clean up cache on old coordinating node.
+                    stateManager.removeBackpressureCounter(detectorId);
+                    featureManager.clear(detectorId);
+                    entityColdStarter.removeDoorKeeper(detectorId);
```

Comment:
Entity cold starter runs on work nodes, hourly cron will clean up expired cache. Will not clean up door keeper here.